src/lib/util/validator.js

```diff
@@ -0,0 +1,15 @@
+const nameRegex = /^[\w.-]+$/g;
+const subscriptionNameRegex = /^[\w.-]+:?[\w.-]+$/g;
+
+const validateWithPattern = (name, pattern) => {
+  if (name && !name.search) {
+    return false;
+  }
+
+  return name.search(pattern) >= 0;
```

Comment:
I recommend  `regex.test(string)` instead of `string.search(regex)` to save from inadvertently calling `search` on something that happens to implement it

Reply:
This was my first instinct as well, however, I ran into some weirdness in the design of the `test` method.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/test#Description

> As with exec() (or in combination with it), test() called multiple times on the same global regular expression instance will advance past the previous match.

This weird side-effect meant that multiple calls to `validate`- when the value hadn't changed- resulted different results.

```
const pattern = /^[\w-.]+$/g
const myStr = "test";

pattern.test(myStr); // true
pattern.test(myStr); // false
pattern.test(myStr); // true
pattern.test(myStr); // false
```

Reply:
Then again... I don't need the global flag.